src/index.js

```diff
@@ -127,40 +130,69 @@ const utils = require('./get-labels');
 
 
     // get labels
-    const labelsFiles = await utils.getLabelsFiles(changedFiles, labelFilename);
-    const labelsFromFiles = await utils.getLabelsFromFiles(labelsFiles);
+    const labelsFiles = await utils.getLabelsFiles(changedFiles, source);
+    const labels = await utils.getLabelsFromFiles(labelsFiles);
 
-    const labelsToRemove = labeledByTheAction.filter((label) => {
-        return !labelsFromFiles.includes(label);
-    });
+    if (!labels.length) {
+        process.exit(0);
+    }
 
-    const labelsToAdd = labelsFromFiles.filter((label) => {
-        return !labeledByTheAction.includes(label);
-    });
+    if (matrixInput) {
+        const matrix = JSON.parse(matrixInput);
 
-    core.info(`labels assigned to pull-request: ${labelsOnPr}`);
-    core.info(`labels which were added by the action: ${labeledByTheAction}`);
-    core.info(`labels to remove: ${labelsToRemove}`);
-    core.info(`labels to add: ${labelsToAdd}`);
+        const output = Object.keys(matrix).reduce((result, entity) => {
+            const prefix = `${entity}:`;
 
-    // add labels
-    if (labelsToAdd.length > 0) {
-        await octokit.rest.issues.addLabels({
-            ...repo,
-            issue_number: pull_request.number,
-            labels: labelsToAdd,
-      });
+            const list = labels
+                .filter((label) => label.includes(prefix))
+                .map((label) => label.split(prefix)[1]);
+
+            if (!list.length) {
+                return result;
+            }
+
+            result = {...result};
+
+            if (list.includes('common')) {
+                result[entity] = matrix[entity];
```

Comment:
I don't like this, it's making assumptions about the code. The action was agnostic until now and I think we shall keep it like this

Reply:
A better approach can be to use labeling in a more micro way and start labeling features/components which are being used in diff projects